packages/block-editor/src/components/rich-text/index.js

```diff
@@ -139,7 +143,8 @@ export function RichTextWrapper(
 			isSelected =
 				selectionStart.clientId === clientId &&
 				selectionEnd.clientId === clientId &&
-				selectionStart.attributeKey === identifier;
+				( selectionStart.attributeKey === identifier ||
+					selectionStart[ instanceIdKey ] === instanceId );
```

Comment:
This condition should also check for undefined `attributeKey`/`undefined`. If a block has two `RichText` components without an identifier, with different instance IDs, the condition will say that both are selected, because `undefined` === `undefined`. But only the one with matching instance ID should be selected.

Reply:
Actually we better not do that because you might want to set selection with just the attribute keys. So this is only for back compat.